server/src/packages/rwcds-form/rwcds-form.service.ts

```diff
@@ -74,4 +126,40 @@ export class RwcdsFormService {
 
     return rwcdsForm;
   }
+
+  // The syncActions method queries for project actions (dcp_projectaction) associated
+  // with the rwcds form's related project. 
+  // Then, it posts each of the project actions to dcp_affectedzoningresolution.
+  // This occurs every time that the rwcds-form endpoint is hit
+  async syncActions(rwcdsForm) {
+    const {
+      dcp_rwcdsform_dcp_affectedzoningresolution_rwcdsform: affectedZoningResolutions,
+      _dcp_projectid_value,
+      dcp_rwcdsformid,
+    } = rwcdsForm;
+    const zrTypes = affectedZoningResolutions.map(zr => zr['dcp_zoningresolutiontype@OData.Community.Display.V1.FormattedValue']);
+    const { records: projectActions } = await this.crmService.get(`dcp_projectactions`, `
+      $filter=_dcp_project_value eq ${_dcp_projectid_value}
+    `);
+
+    await Promise.all(projectActions.map(action => {
+      const projectActionLabel = action['_dcp_action_value@OData.Community.Display.V1.FormattedValue'];
+
+      // Lookup: dcp_affectedzoningresolutions asks for a ZR type,
+      const [actionLabel, actionCode] = ZONING_RESOLUTION_TYPES.find((label) => label === projectActionLabel) || ['ZA', 717170000];
```

Comment:
I might have stored this lookup data a little differently. I think an object would be just fine:

```
const ZONING_RESOLUTION_TYPES = {
  ZA: 717170000,
  // others
};
```

Then,

```
const [actionLabel, actionCode] = ZONING_RESOLUTION_TYPES[projectActionLabel] || ZONING_RESOLUTION_TYPES.ZA;
```

That way we don't have to loop or put constants inside the code after the `||` pipes.

Also, `projectActionLabel` is the lookup key as well as the label, so it doesn't seem worth it to have another value.